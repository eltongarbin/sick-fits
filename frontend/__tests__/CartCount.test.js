import { render, screen } from "@testing-library/react";

import CartCount from "../components/CartCount";

describe("<CartCount />", () => {
  it("renders", () => {
    render(<CartCount count={10} />);
  });

  it("matches snapshot", () => {
    const { container } = render(<CartCount count={11} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("updates via props", () => {
    const { container, rerender } = render(<CartCount count={11} />);
    expect(screen.getByText("11")).toBeInTheDocument();

    rerender(<CartCount count={12} />);
    expect(screen.getByText("12")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
