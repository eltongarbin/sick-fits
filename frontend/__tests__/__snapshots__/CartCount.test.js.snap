// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> matches snapshot 1`] = `
<span
  class="CartCount__AnimationStyles-sc-xxvp4g-1 lcWlXV"
>
  <div>
    <div
      class="CartCount__Dot-sc-xxvp4g-0 hMpcbO count"
    >
      11
    </div>
  </div>
</span>
`;

exports[`<CartCount /> updates via props 1`] = `
<span
  class="CartCount__AnimationStyles-sc-xxvp4g-1 lcWlXV"
>
  <div>
    <div
      class="CartCount__Dot-sc-xxvp4g-0 hMpcbO count count-enter count-enter-active"
    >
      12
    </div>
    <div
      class="CartCount__Dot-sc-xxvp4g-0 hMpcbO count count-exit count-exit-active"
    >
      11
    </div>
  </div>
</span>
`;
